src/serializer/ResidualHeapSerializer.js

```diff
@@ -1515,10 +1560,10 @@ export class ResidualHeapSerializer {
       }
     }
 
-    invariant(
-      this.serializedValues.size === this.residualValues.size,
-      "serialized " + this.serializedValues.size + " of " + this.residualValues.size
-    );
+    // invariant(
```

Comment:
This invariant detects if there is a mismatch between...
- What the visitor things is reachable in the final heap, and
- What the serializer actually serializes.

The serializer should never process values that didn't get visited (because then they didn't get attributed to the right scope), and the visitor should never visit values that don't get serialized (because that indicates that it is visiting "spurious" roots, pulling in and processing stuff that doesn't actually matter, but causes the serializer to emit suboptimal code).

Reply:
Generally, if this invariant fails, it means that there's a mismatch between the visitor and serializer, and that mismatch should be fixed.

Reply:
Feel free to embed a comment to this end, as this questions comes up often.